gcp_variant_transforms/beam_io/vcf_parser.py

```diff
@@ -464,3 +469,179 @@ def _get_variant_calls(self, record, formats):
+               range_tracker,  # type: range_trackers.OffsetRangeTracker
+               file_pattern,  # type: str
+               compression_type,  # type: str
+               allow_malformed_records,  # type: bool
+               representative_header_lines=None,  # type:  List[str]
+               **kwargs  # type: **str
+              ):
+    # type: (...) -> None
+    super(NucleusParser, self).__init__(file_name,
+                                        range_tracker,
+                                        file_pattern,
+                                        compression_type,
+                                        allow_malformed_records,
+                                        representative_header_lines,
+                                        **kwargs)
+    # This member will be properly initiated in _init_with_header().
+    self._vcf_reader = None
+    # These members will be properly initiated in _extract_header_fields().
+    self._is_info_repeated = {}
+    self._is_format_repeated = {}
+    # This is a temporary solution until from_string will be fixed.
+    self._temp_local_file = self._write_to_local_file(file_name)
+
+  def _write_to_local_file(self, remote_file_name):
+    (temp_file, temp_file_name) = tempfile.mkstemp(text=True)
+    with FileSystems.open(remote_file_name) as f:
+      while True:
+        line = f.readline()
+        if line:
+          os.write(temp_file, line)
+        else:
+          os.close(temp_file)
+          break
+    return temp_file_name
+
+  def _store_to_temp_local_file(self, header_lines):
+    (temp_file, temp_file_name) = tempfile.mkstemp(text=True)
+    for line in header_lines:
+      os.write(temp_file, line)
+    os.close(temp_file)
+    return temp_file_name
+
+  def _init_with_header(self, header_lines):
+    # This header line is needed by Nucleus.
+    header_lines = ['##fileformat=VCFv4.2'] + header_lines
+    try:
+      # This is a temporary solution until from_string will be fixed.
+      self._vcf_reader = nucleus.io.vcf.VcfReader(
+          self._temp_local_file, use_index=False)
+
+    except SyntaxError as e:
+      raise ValueError(
+          'Invalid VCF header in %s: %s' % (self._file_name, str(e)))
+    self._extract_header_fields()
+
+  def _extract_header_fields(self):
+    header = self._vcf_reader.header
+    for info in header.infos:
+      if info.number in ('0', '1'):
+        self._is_info_repeated[info.id] = False
+      else:
+        self._is_info_repeated[info.id] = True
+
+    for format_info in header.formats:
+      if format_info.number in ('0', '1'):
+        self._is_format_repeated[format_info.id] = False
+      else:
+        self._is_format_repeated[format_info.id] = True
+
+  def _get_variant(self, data_line):
+    try:
+      # This is a temporary solution until from_string will be fixed.
+      record = next(self._vcf_reader)
+      return self._convert_to_variant(record)
+    except (LookupError, ValueError) as e:
```

Comment:
Again, we should determine what errors Nucleus throws in case of invalid records and adjust accordingly.

Reply:
I actually don't see `LookupError` in their code. Looks like they only throw `ValueError` in their from_string signature.